### PR TITLE
Bug 1190539 - escape HTML in l10n-arguments by default

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1190539

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=643f38f24ea01608b37a5f62bc97ec81b3fca0f5

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c4cb681e5beeb8791c8fddc83e4f6e05337ca46

File: shared/js/l10n.js [hunk 1]
```diff
@@ -1932,13 +1942,20 @@
...
       return false;
     }
 
-    var entity = this.ctx.getEntity(l10n.id, l10n.args);
+    var l10nArgs = element.getAttribute('data-l10n-args');
+
+    var entity = l10nArgs ?
+      this.ctx.getEntity(
+        l10nId,
+        JSON.parse(l10nArgs.replace(reHtml, escapeL10nArgs))
```
Comment (stasm) [inline shared/js/l10n.js]: Could you move the `l10nArgs ?` check into the second argument of `getEntity`?  It can be `undefined` no problem.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=49a6421920af4068ed0cd7ad0c477388428ad0ea

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3db36ebd182ad884ebaaab757075e4232b24a3ae

**Pull request merged**